scripts/peer/peer.py

```diff
@@ -0,0 +1,273 @@
+    sampleid = scores_df.s
+    # only get the first 4 PCs, as indicated by scree plot
+    scores = pd.DataFrame(scores_df['scores'].to_list()).iloc[:, 0:4]
+    scores.columns = ['PC1', 'PC2', 'PC3', 'PC4']
+    scores.insert(loc=0, column='sampleid', value=sampleid)
+    # filter to only CPG samples
+    scores = scores[scores.sampleid.str.contains('CPG')]
+    print(len(scores.index))
+    # 1006
+    # change CPG IDs to One1K1K IDs
+    sampleid_keys = pd.read_csv(sample_id_keys_path, sep=',')
+    scores = pd.merge(
+        scores, sampleid_keys, how='left', left_on='sampleid', right_on='CPG_ID'
+    )
+    print(len(scores.OneK1K_ID.dropna()))
+    # 937
+    # 69 samples are missing RNA-seq data
+    # remove samples which don't have a OneK1K ID and only keep PCA scores
+    scores = scores[~scores.OneK1K_ID.isna()][
+        ['PC1', 'PC2', 'PC3', 'PC4', 'OneK1K_ID']
+    ]
+    # Merge PCA scores with Seyhan's covariate data and drop Seyhan's RNA-seq pca scores
+    # NB: these are the ones in lowercase (e.g., 'pc1')
+    covariates = pd.read_csv(covariates_path, sep='\t')
+    covariates = pd.merge(
+        scores, covariates, how='left', left_on='OneK1K_ID', right_on='sampleid'
+    ).drop(['pc1', 'pc2', 'pc3', 'pc4'], axis=1)
+    # check whether there's any missing data in the covariate data
+    print(covariates[covariates.sampleid.isna()])
+    # OneK1K ID 26_26 does not have any covariate data. Remove this sample and drop
+    # the OneK1K_ID, since it's redundant with sampleid
+    covariates = covariates[~covariates.sampleid.isna()].drop(
+        ['OneK1K_ID'], axis=1
+    )
+    # Match expression data to covariate data, since PEER needs expr and covs
+    # dfs to be the same length
+    expression = pd.read_csv(expression_file, sep='\t')
+    merged_expr_covs = pd.merge(
+        expression, covariates, how='right', left_on='sampleid', right_on='sampleid'
+    )
+    print(merged_expr_covs.shape[0] - merged_expr_covs.dropna().shape[0])
+    # 15 samples with no expression data, but do have covariate and genotype data
+    # This differs per celltype
+    # remove any rows with NA values
+    merged_expr_covs = merged_expr_covs.dropna()
+    expression = merged_expr_covs.drop(
+        ['PC1', 'PC2', 'PC3', 'PC4', 'sex', 'age'], axis=1
+    )
+    covariates = merged_expr_covs[
+        ['sampleid', 'PC1', 'PC2', 'PC3', 'PC4', 'sex', 'age']
+    ]
+    # make sure samples in covariates and expression data are equal
+    expression.sampleid.equals(covariates.sampleid)
+    # True
+    # remove sampleid from both covariates and expression dfs
+    expression.drop('sampleid', axis=1, inplace=True)
+    covariates.drop('sampleid', axis=1, inplace=True)
+    # finally, make sure age and sex in covariate df are int
+    covariates[['sex', 'age']] = covariates[['sex', 'age']].astype(int)
+
+    # return expression data and covariates
+    return covariates.to_csv(index=False), expression.to_csv(index=False)  # type: ignore
```

Comment:
Can we remove the type ignore statement - the initial type annotation in the function is incorrect (it should be Tuple[str, str]). Otherwise can we add a comment on why it's ignored?

Reply:
I've removed the type ignore statement after updating the return type